### PR TITLE
[FIX] xlsx: accelerate xlsx export

### PR DESCRIPTION
**Pull request by rrahir**
This Pull request addresses the slowness of the `pushElement` helper.
The two main improvments concern the iteration over the array elements and the comparison operations.

## Benchmark of `getXLSX` function

On a 30k random string cells (heavy use of `pushElement`) averaged on 10 runs

|         | Firefox     | Chrome |
|-----------------|-----------|------------|
| Before | 153 s    |    81 s     |
| After      | 6.2 s   |   4.8 s    |

With the same patch applied to the mater branch on RNG's super sheet

|         | Firefox     | Chrome |
|-----------------|-----------|------------|
| Before | 351 s    |    95 s     |
| After      | 17.4 s   |   8 s    |

## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [3733743](https://www.odoo.com/web#id=3733743&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/xlsx/helpers/content_helpers.ts [hunk 1]
```diff
@@ -153,21 +138,17 @@ export function addRelsToFile(
...
-    propertyList.push(property);
-    elemId = propertyList.length - 1;
-  }
-  return {
-    id: elemId,
-    list: propertyList,
-  };
+
+  propertyList[propertyList.length] = property;
+  return propertyList.length - 1;
```
Comment (anhe-odoo) [inline src/xlsx/helpers/content_helpers.ts:151]: Can't we re-use `len` here ?
```js
propertyList[len] = property;
return len;
```
?

**Review by anhe-odoo — CHANGES_REQUESTED** [review]

File: tests/__snapshots__/xlsx.test.ts.snap [hunk 2]
```diff
@@ -8001,57 +8001,57 @@ Object {
         </border>
         <border>
             <left/>
+            <right style=\\"thin\\" color=\\"000000\\"/>
+            <top/>
+            <bottom/>
+            <diagonal/>
+        </border>
```
Comment (LucasLefevre) [inline tests/__snapshots__/xlsx.test.ts.snap:8008]: Do you know what causes theses changes in the snapshot ?

**Review by LucasLefevre — COMMENTED** [review]
Nice ! :+1:

**Review by LucasLefevre — APPROVED** [review]
robodoo r+ rebase-ff

**Pull request closed without merging**